File: ArcanAI/Models/Conversation.swift
//
//  Conversation.swift
//  ArcanAI
//
//  Chat conversation management
//

import Foundation

struct Conversation: Identifiable, Codable {
    let id: UUID
    var messages: [Message]
    var modelId: String
    var title: String
    let createdAt: Date
    var updatedAt: Date

    init(id: UUID = UUID(), messages: [Message] = [], modelId: String, title: String = "New Chat", createdAt: Date = Date(), updatedAt: Date = Date()) {
        self.id = id
        self.messages = messages
        self.modelId = modelId
        self.title = title
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }

    mutating func addMessage(_ message: Message) {
        messages.append(message)
        updatedAt = Date()

        // Auto-generate title from first user message
        if title == "New Chat", message.role == .user, !message.content.isEmpty {
            title = String(message.content.prefix(50))
        }
    }
}

File: src/types/conversation.ts
import { Message } from './message';

export interface Conversation {
  id: string;
  title: string;
  messages: Message[];
  modelId?: string;
  createdAt: number;
  updatedAt: number;
}

export function createConversation(modelId?: string, title = 'New chat'): Conversation {
  const now = Date.now();
  return {
    id: `conv-${now}-${Math.random().toString(36).slice(2, 8)}`,
    title,
    messages: [],
    modelId,
    createdAt: now,
    updatedAt: now,
  };
}
